src/hotspot/share/opto/type.cpp

```diff
@@ -473,44 +473,47 @@ void Type::Initialize_shared(Compile* current) {
+  TypeInt::CC_GE    = TypeInt::make( 0, 1, WidenMin);  // == TypeInt::BOOL
+  TypeInt::BYTE     = TypeInt::make(-128, 127,     WidenMin); // Bytes
+  TypeInt::UBYTE    = TypeInt::make(0, 255,        WidenMin); // Unsigned Bytes
+  TypeInt::CHAR     = TypeInt::make(0, 65535,      WidenMin); // Java chars
+  TypeInt::SHORT    = TypeInt::make(-32768, 32767, WidenMin); // Java shorts
+  TypeInt::NON_ZERO = TypeInt::try_make(TypeIntPrototype<jint, juint>{{min_jint, max_jint}, {1, max_juint}, {0, 0}}, WidenMin)->is_int();
+  TypeInt::POS      = TypeInt::make(0, max_jint,   WidenMin); // Non-neg values
+  TypeInt::POS1     = TypeInt::make(1, max_jint,   WidenMin); // Positive values
+  TypeInt::INT      = TypeInt::make(min_jint, max_jint, WidenMax); // 32-bit integers
+  TypeInt::SYMINT   = TypeInt::make(-max_jint, max_jint, WidenMin); // symmetric range
+  TypeInt::TYPE_DOMAIN = TypeInt::INT;
   // CmpL is overloaded both as the bytecode computation returning
-  // a trinary (-1,0,+1) integer result AND as an efficient long
+  // a trinary (-1, 0, +1) integer result AND as an efficient long
   // compare returning optimizer ideal-type flags.
-  assert( TypeInt::CC_LT == TypeInt::MINUS_1, "types must match for CmpL to work" );
-  assert( TypeInt::CC_GT == TypeInt::ONE,     "types must match for CmpL to work" );
-  assert( TypeInt::CC_EQ == TypeInt::ZERO,    "types must match for CmpL to work" );
-  assert( TypeInt::CC_GE == TypeInt::BOOL,    "types must match for CmpL to work" );
-  assert( (juint)(TypeInt::CC->_hi - TypeInt::CC->_lo) <= SMALLINT, "CC is truly small");
-
-  TypeLong::MAX = TypeLong::make(max_jlong);  // Long MAX
-  TypeLong::MIN = TypeLong::make(min_jlong);  // Long MIN
-  TypeLong::MINUS_1 = TypeLong::make(-1);        // -1
-  TypeLong::ZERO    = TypeLong::make( 0);        //  0
-  TypeLong::ONE     = TypeLong::make( 1);        //  1
-  TypeLong::POS     = TypeLong::make(0,max_jlong, WidenMin); // Non-neg values
-  TypeLong::LONG    = TypeLong::make(min_jlong,max_jlong,WidenMax); // 64-bit integers
-  TypeLong::INT     = TypeLong::make((jlong)min_jint,(jlong)max_jint,WidenMin);
-  TypeLong::UINT    = TypeLong::make(0,(jlong)max_juint,WidenMin);
-  TypeLong::TYPE_DOMAIN  = TypeLong::LONG;
+  assert(TypeInt::CC_LT == TypeInt::MINUS_1, "types must match for CmpL to work" );
+  assert(TypeInt::CC_GT == TypeInt::ONE,     "types must match for CmpL to work" );
+  assert(TypeInt::CC_EQ == TypeInt::ZERO,    "types must match for CmpL to work" );
+  assert(TypeInt::CC_GE == TypeInt::BOOL,    "types must match for CmpL to work" );
+
+  TypeLong::MAX = TypeLong::make(max_jlong); // Long MAX
+  TypeLong::MIN = TypeLong::make(min_jlong); // Long MIN
+  TypeLong::MINUS_1  = TypeLong::make(-1);   // -1
+  TypeLong::ZERO     = TypeLong::make( 0);   //  0
+  TypeLong::ONE      = TypeLong::make( 1);   //  1
+  TypeLong::NON_ZERO = TypeLong::try_make(TypeIntPrototype<jlong, julong>{{min_jlong, max_jlong}, {1, max_julong}, {0, 0}}, WidenMin)->is_long();
```

Comment:
Here we really expect the `try` not to fail, right? Maybe we could have some `make` method, that calls `try_make` and asserts that we succeed?

Reply:
It is a one-liner so I think making a dedicate API point is unreasonable, normally we don't know `try_make` will return a non-empty set unless the constraints are all constants like in this case.